asteroid/data/avspeech_dataset.py

```diff
@@ -1,25 +1,186 @@
+        '''
+        self.video = augmenter(self.video, *args, **kwargs)
+
+    def _convert_video(self):
+        if self.embed_saved:
+            return
+        self.buffer_video = get_frames(self.video)
+
+    def _check_video_embed(self, embed_ext=".npy"):
+        video_name_stem = self.video_path.stem
+
+        embed_dir = self.embed_dir
+        if not embed_dir.is_dir():
+            embed_dir = Path(*embed_dir.parts[2:])
+
+        self.embed_path = Path(embed_dir, video_name_stem + f"_part{self.video_start_length}" + embed_ext)
+        if self.embed_path.is_file():
+            self.embed_saved = True
+            self.embed = np.load(self.embed_path.as_posix())
+
+    def embed_is_saved(self):
+        return self.embed_saved
+
+    def get_embed(self):
+        return self.embed
+
+    def get_video(self):
+        #retrieve slice of video, if video > 75 frames
+        buffer_video = self.buffer_video[self.video_start_length*75: (self.video_start_length+1)*75]
+        return buffer_video
+
+    def get_audio(self):
+        return self.audio
+
+    def get_spec(self):
+            return np.load(self.spec_path)
+
+    def is_spec(self):
+        return self._is_spec
+
+    @staticmethod
+    def load_audio(audio_path: str, sr=16_000, load_spec=False):
+        audio_path = Path(audio_path)
+        spec_exists = False
+        spec_path = Path(*audio_path.parts[:-2], "spec", audio_path.stem + ".npy")
+        if load_spec and spec_path.is_file():
+            spec_exists = True
+            audio = np.load(spec_path)
+        else:
+            audio = librosa.load(str(audio_path), sr=sr)[0]
+        return audio, spec_exists, spec_path
+
 
 class AVSpeechDataset(data.Dataset):
 
     def __init__(self, input_df_path: Path,
-                signal_constructor, convert_to_spectrogram,
-                input_audio_size=2):
+                 embed_dir: Path,
+                 input_audio_size=2):
         """
 
             Args:
                 input_df_path: path for combination dataset
                 input_audio_size: total audio/video inputs
         """
         self.input_audio_size = input_audio_size
+        self.embed_dir = embed_dir
         self.input_df = pd.read_csv(input_df_path.as_posix())
-        self.signal_constructor = signal_constructor
-        self.convert_to_spectrogram = convert_to_spectrogram
+
+    @staticmethod
+    def encode(x: np.ndarray, p=0.3, stft_encoder=None):
+        if stft_encoder is None:
+            stft_encoder = Encoder(STFTFB(n_filters=512, kernel_size=400,
+                                          stride=160))
+
+        x = torch.from_numpy(x).float()
+
+        # time domain to time-frequency representation
+        tf_rep = stft_encoder(x).squeeze(0).unsqueeze(2)
+        # power law on complex numbers
+        tf_rep = (torch.abs(tf_rep) ** p) * torch.sign(tf_rep)
+        # (514, 298) -> (257, 298, 2)
+        tf_rep = torch.cat((tf_rep[:257], tf_rep[257:]), axis=2)
```

Comment:
We have helper functions `transforms.from_torchaudio`, `transforms.to_torchaudio`. This will avoid hardcoding the 257 frequencies.

Reply:
Oh, okay. Will do that.